### PR TITLE
Tslint updater

### PR DESCRIPTION
**Pull request by gabritto**
This PR creates a script that runs TSLint for DT packages, taking as configuration `dt.json` and ignoring the existing `tslint.json` for the package. The output is a new `tslint.json` for the packages, that only adds exemptions for rules that actually produced lint failures.

### CODE REVIEW

File: src/updateConfig.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
+import cp = require("child_process");
```
Comment (sandersn) [inline src/updateConfig.ts]: can you add a comment that explains that this is a stand-alone script and that it's intended to make tslint.json have exactly the correct set of rule exemptions? (at least, that is what I think it does -- if I'm wrong, take that as something that definitely needs to be explained)

Comment (gabritto) [reply src/updateConfig.ts]: see if the comment i added makes sense or if there's still some things that are unclear.

File: src/updateConfig.ts [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+ * Represents a package from the linter's perspective.
+ * For example, `DefinitelyTyped/types/react` and `DefinitelyTyped/types/react/v15` are different
+ * packages.
+ */
+class LintPackage {
+    private rootDir: string;
+    private files: ts.SourceFile[];
+    private program: ts.Program;
+
+    constructor(rootDir: string) {
```
Comment (sandersn) [inline src/updateConfig.ts]:
```suggestion
    constructor(private rootDir: string) {
```

and then you can drop the property declaration and `this.rootDir = rootDir`

File: src/updateConfig.ts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+}
+
+/**
+ * Represents a package from the linter's perspective.
+ * For example, `DefinitelyTyped/types/react` and `DefinitelyTyped/types/react/v15` are different
+ * packages.
+ */
+class LintPackage {
+    private rootDir: string;
+    private files: ts.SourceFile[];
```
Comment (sandersn) [inline src/updateConfig.ts]:
```suggestion
    private files: ts.SourceFile[] = [];
```

File: src/updateConfig.ts [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+            ruleToFailures.set(failureJson.ruleName, [failureJson]);
+        }
+    }
+
+    const newRulesConfig: Config.RawRulesConfig = {};
+    ruleToFailures.forEach((failures, rule) => {
+        if (ignoredRules.includes(rule)) {
+            return;
+        }
+        const disabler = ruleDisablers.get(rule) || defaultDisabler;
```
Comment (sandersn) [inline src/updateConfig.ts]: I would personally hard-code this with a conditional instead of a Map+default since there's only one complex rule right now.

**Pull request merged**